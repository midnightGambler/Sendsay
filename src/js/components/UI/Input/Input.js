import React from "react";

const Input = ({ label, className = '', error, ...rest }) => (
  <div className={`input ${className}`}>
    <label className="input__label">{label}</label>
    <input className="input__input" {...rest} />
    {error && <span className="input-error">{error}</span>}
  </div>
);

export default Input;
