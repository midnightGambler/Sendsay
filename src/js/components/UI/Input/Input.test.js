import React from "react";
import { shallow } from "enzyme";
import Input from "./Input";

it("should render Input field properly", () => {
  const input = shallow(<Input label="test" />);
  expect(input).toMatchSnapshot();
});

it("should render Input field with error", () => {
  const input = shallow(<Input label="test" error="error" />);
  expect(input).toMatchSnapshot();
});

it('should test Input onChange event', () => {
  const mockedCallback = jest.fn()
  const input = shallow(<Input onChange={mockedCallback} label="test" />);
  input.find('input').simulate('change');
  expect(mockedCallback.mock.calls.length).toBe(1);
})
