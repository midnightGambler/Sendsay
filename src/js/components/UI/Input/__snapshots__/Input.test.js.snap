// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Input field properly 1`] = `
<div
  className="input "
>
  <label
    className="input__label"
  >
    test
  </label>
  <input
    className="input__input"
  />
</div>
`;

exports[`should render Input field with error 1`] = `
<div
  className="input "
>
  <label
    className="input__label"
  >
    test
  </label>
  <input
    className="input__input"
  />
  <span
    className="input-error"
  >
    error
  </span>
</div>
`;
